From 5928c79092044416cff058848805af24c2bbaeb4 Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Tue, 4 Mar 2025 18:14:46 +0100
Subject: [PATCH 1620/1695] clk: stm32mp21: remove CCB clock

CCB peripheral is no more supported.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 1a4b7f232ffadf1aabd189e0ccc66c89f2ac3988 clk: stm32mp21: remove CCB clock]
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Change-Id: I0962468f8c4932839610403610d2f5b64167fcbd
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/441746
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/drivers/clk/clk-stm32mp21.c     | 4 ----
 core/include/drivers/stm32mp21_rcc.h | 6 ------
 2 files changed, 10 deletions(-)

diff --git a/core/drivers/clk/clk-stm32mp21.c b/core/drivers/clk/clk-stm32mp21.c
index 7365ac578..be0a1dbc3 100644
--- a/core/drivers/clk/clk-stm32mp21.c
+++ b/core/drivers/clk/clk-stm32mp21.c
@@ -368,7 +368,6 @@ enum enum_gate_cfg {
 	GATE_HASH2,
 	GATE_CRYP1,
 	GATE_CRYP2,
-	GATE_CCB,
 	GATE_IWDG1,
 	GATE_IWDG2,
 	GATE_IWDG3,
@@ -550,7 +549,6 @@ static const struct gate_cfg gates_mp21[GATE_NB] = {
 	GATE_CFG(GATE_HASH2,		RCC_HASH2CFGR,		1,	0),
 	GATE_CFG(GATE_CRYP1,		RCC_CRYP1CFGR,		1,	0),
 	GATE_CFG(GATE_CRYP2,		RCC_CRYP2CFGR,		1,	0),
-	GATE_CFG(GATE_CCB,		RCC_CCBCFGR,		1,	0),
 	GATE_CFG(GATE_IWDG1,		RCC_IWDG1CFGR,		1,	0),
 	GATE_CFG(GATE_IWDG2,		RCC_IWDG2CFGR,		1,	0),
 	GATE_CFG(GATE_IWDG3,		RCC_IWDG3CFGR,		1,	0),
@@ -3639,7 +3637,6 @@ static STM32_GATE(ck_icn_p_cryp1, &ck_icn_ls_mcu, 0, GATE_CRYP1);
 static STM32_GATE(ck_icn_p_cryp2, &ck_icn_ls_mcu, 0, GATE_CRYP2);
 static STM32_GATE(ck_icn_p_saes, &ck_icn_ls_mcu, 0, GATE_SAES);
 static STM32_GATE(ck_icn_p_pka, &ck_icn_ls_mcu, 0, GATE_PKA);
-static STM32_GATE(ck_icn_p_ccb, &ck_icn_ls_mcu, 0, GATE_CCB);
 static STM32_GATE(ck_icn_p_eth1, &ck_icn_ls_mcu, 0, GATE_ETH1);
 static STM32_GATE(ck_icn_p_eth2, &ck_icn_ls_mcu, 0, GATE_ETH2);
 static STM32_GATE(ck_icn_p_adc1, &ck_icn_ls_mcu, 0, GATE_ADC1);
@@ -3945,7 +3942,6 @@ static struct clk *stm32mp21_clk_provided[STM32MP21_ALL_CLK_NB] = {
 	[CK_BUS_CRYP2]		= &ck_icn_p_cryp2,
 	[CK_BUS_SAES]		= &ck_icn_p_saes,
 	[CK_BUS_PKA]		= &ck_icn_p_pka,
-	[CK_BUS_CCB]		= &ck_icn_p_ccb,
 	[CK_BUS_GPIOA]		= &ck_icn_p_gpioa,
 	[CK_BUS_GPIOB]		= &ck_icn_p_gpiob,
 	[CK_BUS_GPIOC]		= &ck_icn_p_gpioc,
diff --git a/core/include/drivers/stm32mp21_rcc.h b/core/include/drivers/stm32mp21_rcc.h
index 39e29b5ab..c072ad876 100644
--- a/core/include/drivers/stm32mp21_rcc.h
+++ b/core/include/drivers/stm32mp21_rcc.h
@@ -392,7 +392,6 @@
 #define RCC_HASH2CFGR				U(0x884)
 #define RCC_CRYP1CFGR				U(0x888)
 #define RCC_CRYP2CFGR				U(0x88C)
-#define RCC_CCBCFGR				U(0x890)
 #define RCC_IWDG1CFGR				U(0x894)
 #define RCC_IWDG2CFGR				U(0x898)
 #define RCC_IWDG3CFGR				U(0x89C)
@@ -1595,11 +1594,6 @@
 #define RCC_CRYP2CFGR_CRYP2EN			BIT(1)
 #define RCC_CRYP2CFGR_CRYP2LPEN			BIT(2)
 
-/* RCC_CCBCFGR register fields */
-#define RCC_CCBCFGR_CCBRST			BIT(0)
-#define RCC_CCBCFGR_CCBEN			BIT(1)
-#define RCC_CCBCFGR_CCBLPEN			BIT(2)
-
 /* RCC_IWDG1CFGR register fields */
 #define RCC_IWDG1CFGR_IWDG1EN			BIT(1)
 #define RCC_IWDG1CFGR_IWDG1LPEN			BIT(2)
-- 
2.47.2

